docs/how_to_publish_to_cmr_with_earthdata_varinfo.ipynb

```diff
@@ -0,0 +1,314 @@
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Saves granule to current directory or\n",
+    "# out_directory=/usr/given/path\n",
+    "download_granule(link, auth_header=auth_header)"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Instantiate a ```VarInfoFromNetCDF4``` object for a local NetCDF-4 file. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 17,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Instantiate a VarInfoFromNetCDF4 object for a local NetCDF-4 file.\n",
+    "var_info = VarInfoFromNetCDF4('MERRA2_400.inst1_2d_asm_Nx.20220130.nc4',\n",
+    "                              short_name='M2I1NXASM')"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Retrieve a dictionary of UMM-Var JSON records"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Keys are the full variable paths, values are \n",
+    "# UMM-Var schema-compatible, JSON-serialisable dictionaries.\n",
+    "umm_var_dict = get_all_umm_var(var_info)\n",
+    "umm_var_dict"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Publish all the UMM-Var entries for M2I1NXASM"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "publish_all_umm_var('C1256535511-EEDTEST',\n",
+    "                    umm_var_dict,\n",
+    "                    auth_header=auth_header,\n",
+    "                    cmr_env=CMR_UAT)"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Publish a single UMM-Var entry"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Keys are the full variable paths, values are \n",
+    "# UMM-Var schema-compatible, JSON-serialisable dictionaries.\n",
+    "umm_var_entry = get_umm_var(var_info, var_info.get_variable('/TROPPV'))\n",
+    "umm_var_entry"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Retrieve a dictionary of a single UMM-Var JSON records. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "publish_umm_var('C1256535511-EEDTEST',\n",
+    "                umm_var_entry,\n",
+    "                auth_header=auth_header,\n",
+    "                cmr_env=CMR_UAT)"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Download a granule for the collection **GLDAS_NOAH10_3H**, generate it's UMM-Var entry and publish it to CMR, with our one function: ```generate_collection_umm_var```"
```

Comment:
The flow of this notebook makes a lot of sense from the perspective of the order we did development. But this function is kind of the all-singing-all-dancing thing that we're trying to offer to folks to make their lives as easy as possible. Because of that, I reckon we should have this before all the individual function calls. Sort of the TL;DR workflow, you know?

Nitpick: This is a pretty long heading. I think that probably suggests that the heading could be shorter and the details expanding into some regular-text sentences in the markdown block.

Reply:
Good idea!
* [ ] Switch functions around to one function to rule them all first

Reply:
Thanks for switching things around. And I really like that you have the separate `import` statement that just grabs `generate_collection_umm_var`; it highlights just how much less you need to do that the alternative workflow.